src/Symfony/Component/Serializer/Mapping/AttributeMetadataInterface.php

```diff
@@ -71,4 +71,24 @@ public function getSerializedName(): ?string;
      * Merges an {@see AttributeMetadataInterface} with in the current one.
      */
     public function merge(self $attributeMetadata);
+
+    /**
+     * Sets the version number from which the attribute must be serialized.
+     */
+    public function setSince(string $version);
+
+    /**
+     * Gets the version number from which the attribute must be serialized.
+     */
+    public function getSince(): ?string;
+
+    /**
+     * Sets the version number after which the attribute must not be serialized.
+     */
+    public function setUntil(string $version);
+
+    /**
+     * Gets the version number after which the attribute must not be serialized.
+     */
+    public function getUntil(): ?string;
```

Comment:
theses changes are a bc break (you can't modify and add / remove methods / constants on an interface).

This should be documented in the upgrade file, and on 5.0 these methods should be added on the interface.

Reply:
This interface is internal, so it's allowed.